static/js/publisher/release/reducers/pendingReleases.test.js

```diff
@@ -776,4 +777,86 @@ describe("pendingReleases", () => {
       });
     });
   });
+
+  describe("on CANCEL_PROGRESSIVE_RELEASE action", () => {
+    const cancelProgressiveReleaseAction = {
+      type: CANCEL_PROGRESSIVE_RELEASE,
+      payload: {
+        key: "progressive-test",
+        previousRevision: {
+          architectures: ["amd64"],
+          revision: 2
+        }
+      }
+    };
+
+    describe("when state is empty", () => {
+      const emptyState = {};
+
+      it("should not add a pendingRelease", () => {
+        const result = pendingReleases(
+          emptyState,
+          cancelProgressiveReleaseAction
+        );
+
+        expect(result).toEqual(emptyState);
+      });
+    });
+
+    describe("when there is a non-progressive pendingRelease", () => {
+      const nonProgressiveState = {
+        "1": {
+          "latest/stable": {
+            revision: { revision: 1, architectures: ["amd64"] },
+            channel: "latest/stable"
+          }
+        }
+      };
+
+      it("should not change the pendingRelease", () => {
+        const result = pendingReleases(
+          nonProgressiveState,
+          cancelProgressiveReleaseAction
+        );
+
+        expect(result).toEqual(nonProgressiveState);
+      });
+    });
+
+    describe("when there is a progressive pendingRelease", () => {
```

Comment:
I think there is a missing case when there are multiple releases with same progressive key

Reply:
Added - this also clarifies the expected (if wrong, currently) behaviour a little